core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/AutoscaleV4Endpoint.java

```diff
@@ -106,4 +107,10 @@ void decommissionInstancesForClusterCrn(@PathParam("crn") String clusterCrn,
     @Path("clusterproxy")
     @Produces(MediaType.APPLICATION_JSON)
     ClusterProxyConfiguration getClusterProxyconfiguration();
+
+    @GET
+    @Path("recommendation")
+    @Produces(MediaType.APPLICATION_JSON)
+    AutoscaleRecommendationV4Response getRecommendation(@QueryParam("workspaceId") Long workspaceId,
+                                                        @QueryParam("blueprintName") String blueprintName);
 }
```

Comment:
Can this take only clusterCrn and internally use blueprintName ?

Reply:
Doing it by blueprintName is more cache-friendly. At the same time, clusterCrn is a nice abstraction, particularly caller need not worry about workspaceId.

Reply:
I would prefer it using clusterCrn, since it avoids additional CB call to retrieve blueprintname associated with the cluster. Also workspaceId is internal id of CB, so  would  prefer to avoid making api calls based on that.